src/test/regress/upgrade/custom_citus.py

```diff
@@ -0,0 +1,151 @@
+                   config.random_worker_port(), CUSTOM_SQL_SCHEDULE, config.output_dir, config.input_dir, config.user)
+    else:
+        exitCode |= common.run_pg_regress_without_exit(config.bindir, config.pg_srcdir,
+                   config.coordinator_port(), CUSTOM_SQL_SCHEDULE, config.output_dir, config.input_dir, config.user)
+
+    run_time = time.time() - start_time
+    testResults[name] =  "SUCCESS" if exitCode == 0 else "FAIL: see {}".format(config.output_dir + '/run.out')
+    testResults[name] += " runtime: {} seconds".format(run_time)
+
+    if exitCode != 0:
+        failCount += 1
+
+    common.stop_databases(config.bindir, config.datadir, config.node_name_to_ports)
+    common.save_regression_diff('sql', config.output_dir)
+
+def copy_test_files(config):
+
+    sql_dir_path = os.path.join(config.datadir, 'sql')
+    expected_dir_path = os.path.join(config.datadir, 'expected')
+
+    common.initialize_temp_dir(sql_dir_path)
+    common.initialize_temp_dir(expected_dir_path)
+    for test_name in CUSTOM_TEST_NAMES:
+        sql_name = os.path.join('./sql', test_name + '.sql')
+        output_name = os.path.join('./expected', test_name + '.out')
+        shutil.copy(sql_name, sql_dir_path)
+        shutil.copy(output_name, expected_dir_path)
+
+
+
+class TestRunner(threading.Thread):
+    def __init__(self, config, name):
+      threading.Thread.__init__(self)
+      self.config = config
+      self.name = name
+
+    def run(self):
+      try:
+        run_for_config(self.config, self.name)
+      except Exception as e:
+          print(e)
+
+
+if __name__ == '__main__':
+    docoptRes = docopt(__doc__)
+    configs = [
+        (CitusDefaultClusterConfig(docoptRes), "default citus cluster"),
+        (CitusSingleNodeClusterConfig(docoptRes), "single node citus cluster"),
+        (CitusSingleNodeSingleShardClusterConfig(docoptRes), "single node single shard cluster"),
+        (CitusSingleShardClusterConfig(docoptRes), "single shard multiple workers cluster"),
+        (CitusNonMxClusterConfig(docoptRes), "mx citus cluster"),
+        (CitusManyShardsClusterConfig(docoptRes), "citus cluster with many shards"),
+        (CitusSingleNodeSingleSharedPoolSizeClusterConfig(docoptRes), "citus single node single shared pool cluster"),
+        (CitusSingleNodeSingleConnectionClusterConfig(docoptRes), "citus single node single connection cluster"),
+        (CitusSingleWorkerClusterConfig(docoptRes), "citus single worker node cluster"),
+        (CitusShardReplicationFactorClusterConfig(docoptRes), "citus with shard replication factor 2 cluster"),
+        (CitusNoLocalExecutionClusterConfig(docoptRes), "citus with no local execution cluster"),
+        (CitusComplexClusterConfig(docoptRes), "citus different settings cluster"),
+        (CitusSuperUserDefaultClusterConfig(docoptRes), "citus super user default cluster"),
+
+    ]
+
+    start_time = time.time()
+
+
+    parallel_thread_amount = 1
```

Comment:
Is it a good idea to run this sequentially by default or should we choose something more? Like 5? We saw that we might get "shared memory segment" errors with higher parallelism in some environments

Reply:
I would make it a bit parallel at least by default, like 2 or 3 or maybe 5. At what parallelism did we get these errors you're talking about?

Reply:
I haven't got it personally, but Onder got it with 10 I believe.

Reply:
I think after we added properly closing the databases, this is not an issue on my laptop. Still, lets maybe keep it something reasonable because we already run individual tests in parallel during the schedules.

So, maybe start with 2 or 4 for now?

If possible, it'd probably makes sense to only CI have the parallelism, but not the default experience on local machines. Because (a) the output is super messy in the stdout (b) it puts a lot of pressure to the machine